### PR TITLE
A1 fixes to address stopping behavior and plugin connections 

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
This PR adds a number of fixes to the A1 release candidate. The following changes are made
* Fix incorrect service names in Intersection Transit Maneuvering (ITM) plugin
* Add a check in ITM for service availability so a clear error can be provided
* Remove unused world model dependencies from ITM
* Add numerous debug prints to ITM
* Adjust stop and wait behavior to target the midpoint of the stopping buffer instead of the end
* Increase the default stopping buffers for stop_and_wait and work zone plugins to 5m. 
* Update stop and wait crawl speed to 5mph and the hardcoded planning buffer to 1mph 
* Add some debug logging

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1444

## Motivation and Context

Functional workzone use case
## How Has This Been Tested?
Tested at TFHRC in black pacifica 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: intersection_transit_maneuvering/launch/intersection_transit_maneuvering.launch [hunk 1]
```diff
@@ -17,5 +17,7 @@
 
 <launch>
     <node name="intersection_transit_maneuvering" pkg="intersection_transit_maneuvering" type="intersection_transit_maneuvering">
+      <remap to="plugins/IntersectionTransitPlugin/plan_trajectory" from="plan_trajectory"/>
+       <remap to="plugins/InLaneCruisingPlugin/plan_trajectory" from="plugins/InlaneCruisingPlugin/plan_trajectory"/>
```
Comment (MishkaMN) [inline intersection_transit_maneuvering/launch/intersection_transit_maneuvering.launch:21]: no need for this as it was fixed in the cpp

File: stop_and_wait_plugin/src/stop_and_wait_plugin.cpp [hunk 2]
```diff
@@ -219,7 +219,9 @@ std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_
 
   std::vector<PointSpeedPair> final_points;
 
-  double remaining_distance = stop_location - starting_downtrack;  
+  // TODO think about targeting the middle of the buffer
```
Comment (MishkaMN) [inline stop_and_wait_plugin/src/stop_and_wait_plugin.cpp:222]: remove TODO comment

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: stop_and_wait_plugin/src/stop_and_wait_plugin.cpp [hunk 3]
```diff
@@ -275,31 +294,34 @@ std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_
   std::vector<lanelet::BasicPoint2d> raw_points;
   splitPointSpeedPairs(final_points, &raw_points, &speeds);
 
-  std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(raw_points);
+  // Convert the inverted downtracks back to regular downtracks
+  double max_downtrack = inverse_downtracks.back();
+  std::vector<double> downtracks = lanelet::utils::transform(inverse_downtracks, [max_downtrack](const auto& d) { return max_downtrack - d; });
+  std::reverse(downtracks.begin(),
+               downtracks.end());
```
Comment (MishkaMN) [inline stop_and_wait_plugin/src/stop_and_wait_plugin.cpp:301]: Are we sure that we are not "reverting" them first by changing downtrack values, and the reverting them again by std::reverse functon?

Comment (msmcconnell) [reply stop_and_wait_plugin/src/stop_and_wait_plugin.cpp:301]: Yes. If you work it out by hand, the inverse one has both reversed indexes and inverted counts, so we need to fix the counts then correct the indexes in two steps.

Comment (MishkaMN) [reply stop_and_wait_plugin/src/stop_and_wait_plugin.cpp:300]: ah makes sense

**Review by MishkaMN — APPROVED** [review]

**Pull request merged**